### PR TITLE
Control 'Touch Keyboard and Handwriting Panel Service' warning

### PR DESCRIPTION
**Pull request by WVVxm**
## Summary of the Pull Request
Add a setting to turn off the warning  if 'Touch Keyboard and Handwriting Panel Service' is disabled.
 
## References
#8095
https://github.com/microsoft/terminal/issues/7886#issuecomment-729350169

## PR Checklist
* [ ] Closes #xxx
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
The service might not start in some case, and it doesn't affect the input in some computer.
This PR turn off the warning even if the service is disabled.
The setting name is  "inputServiceWarning".

part of settings.json:

    // If enabled, selections are automatically copied to your clipboard.
    "copyOnSelect": false,
    
    // If disabled, don't show warning if 'Touch Keyboard and Handwriting Panel Service' disabled.
    "inputServiceWarning": false,

## Validation Steps Performed
I manually set the service to "Disabled", restarted the Terminal, verified the warning up, then set "inputServiceWarning" to false and restarted the Terminal, and the warning didn't appear.

### CODE REVIEW

**Comment by DHowett** [comment]
Hey, thanks for writing this up! I'm not as averse to this setting as I was in the linked issues, but we'll need a little bit to review this. It's a busy week for the team -- since 1.6 just came out, we have a bit of a backlog. Thanks for understanding!

File: doc/cascadia/profiles.schema.json [hunk 1]
```diff
@@ -629,6 +629,11 @@
           "description": "When set to true, tabs are always displayed. When set to false and \"showTabsInTitlebar\" is set to false, tabs only appear after opening a new tab.",
           "type": "boolean"
         },
+        "inputServiceWarning": {
```
Comment (Don-Vito) [inline doc/cascadia/profiles.schema.json:632]: Nit: consider renaming into "showInputServiceWarning" (here and in other places)

File: src/cascadia/TerminalApp/AppLogic.cpp [hunk 2]
```diff
@@ -487,11 +487,15 @@ namespace winrt::TerminalApp::implementation
     void AppLogic::_OnLoaded(const IInspectable& /*sender*/,
                              const RoutedEventArgs& /*eventArgs*/)
     {
-        const auto keyboardServiceIsDisabled = !_IsKeyboardServiceEnabled();
-        if (keyboardServiceIsDisabled)
+        if (_settings.GlobalSettings().InputServiceWarning())
```
Comment (Don-Vito) [inline src/cascadia/TerminalApp/AppLogic.cpp:490]: Nit: consider unifying into a single if statement with &&

File: src/cascadia/TerminalSettingsModel/userDefaults.json [hunk 3]
```diff
@@ -16,6 +16,9 @@
     // If enabled, selections are automatically copied to your clipboard.
     "copyOnSelect": false,
 
+    // If disabled, don't show warning if 'Touch Keyboard and Handwriting Panel Service' disabled.
```
Comment (Don-Vito) [inline src/cascadia/TerminalSettingsModel/userDefaults.json:19]: Not sure if we want to add it here.

**Review by Don-Vito — COMMENTED** [review]
I am neither an expert nor from the team, but besides few nits below the change looks good to me.

Probably the next missing pieces are
1. Adding this to Settings UI 
2. Modifying the documentation.

**However** before you do it, probably it worth understanding from the team what are their plans around this feature and https://github.com/microsoft/terminal/pull/8771 that introduces application state. Once that PR lands, we can introduce a UX that will allow the user to "silence" this warning forever, without editing the settings (e.g., by adding "never show this warning again" to the bar).

File: doc/cascadia/profiles.schema.json (hunk 1 above)
Comment (zadjii-msft) [reply doc/cascadia/profiles.schema.json:632]: I'm gonna say that this one is probably okay as is. The other warnings are `largePasteWarning` and `multiLinePasteWarning`, so this is actually consistent.

File: src/cascadia/TerminalControl/IControlSettings.idl [hunk 4]
```diff
@@ -39,6 +39,7 @@ namespace Microsoft.Terminal.TerminalControl
         Microsoft.Terminal.TerminalControl.IKeyBindings KeyBindings;
 
         Boolean CopyOnSelect;
+        Boolean InputServiceWarning;
```
Comment (zadjii-msft) [inline src/cascadia/TerminalControl/IControlSettings.idl:42]: This doesn't actually need to be here. We don't need this setting down in the control layer, so we can omit it here.

File: src/cascadia/TerminalSettingsModel/userDefaults.json (hunk 3 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/userDefaults.json:19]: I concur, we should remove these lines from this file. This file is intentionally fairly bare.

**Review by zadjii-msft — CHANGES_REQUESTED** [review]
Alright so I'm blocking over removing the lines from `IControlSettings` and from `userDefaults.json`, but otherwise this looks fine to me. The other warnings aren't expressed in the settings UI quite yet, so I wouldn't worry about this one either.

**Review by zadjii-msft — APPROVED** [review]

File: src/cascadia/TerminalSettingsModel/defaults-universal.json [hunk 5]
```diff
@@ -7,6 +7,7 @@
     "initialRows": 30,
     "launchMode": "default",
     "alwaysOnTop": false,
+    "inputServiceWarning":true,
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsModel/defaults-universal.json:10]:
```suggestion
```

It is okay if we do not include this setting in the defaults files. Thank you for your attention to detail!

I will commit this and merge your PR -- no need to send another update.

File: src/cascadia/TerminalSettingsModel/defaults.json [hunk 6]
```diff
@@ -7,6 +7,7 @@
     "initialRows": 30,
     "launchMode": "default",
     "alwaysOnTop": false,
+    "inputServiceWarning":true,
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsModel/defaults.json:10]:
```suggestion
```

It is okay if we do not include this setting in the defaults files. Thank you for your attention to detail!

I will commit this and merge your PR -- no need to send another update.

**Review by DHowett — APPROVED** [review]

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
Thank you for your contribution!

**Pull request merged**